scripts/anvil_tools/set_up_anvil_workspaces.py

```diff
@@ -0,0 +1,255 @@
+
+    json_request = json.dumps(acls)
+
+    # request URL for updateWorkspaceACL
+    uri = f"https://api.firecloud.org/api/workspaces/{project}/{workspace_name}/acl?inviteUsersNotFound=false"
+
+    # Get access token and and add to headers for requests.
+    headers = {"Authorization": "Bearer " + get_access_token(), "accept": "*/*", "Content-Type": "application/json"}
+    # -H  "accept: */*" -H  "Authorization: Bearer [token] -H "Content-Type: application/json"
+
+    # capture response from API and parse out status code
+    response = requests.patch(uri, headers=headers, data=json_request)
+    status_code = response.status_code
+
+    emails = [acl['email'] for acl in acls]
+    # print success or fail message based on status code
+    if status_code != 200:
+        print(f"WARNING: Failed to update {project}/{workspace_name} with the following user(s)/group(s): {emails}.")
+        print("Check output file for error details.")
+        return status_code, response.json()
+    else:
+        print(f"Successfully updated {project}/{workspace_name} with the following user(s)/group(s): {emails}.")
+        return status_code, emails
+
+
+def check_workspace_exists(workspace_name, project="anvil-datastorage"):
+    """Determine if a workspace of given namespace/name already exists."""
+
+    # don't need full response - could be very large and time consuming
+    uri = f"https://api.firecloud.org/api/workspaces/{project}/{workspace_name}?fields=owners,workspace.createdBy,workspace.authorizationDomain"
+
+    # Get access token and and add to headers for requests.
+    # -H  "accept: application/json" -H  "Authorization: Bearer [token]
+    headers = {"Authorization": "Bearer " + get_access_token(), "accept": "application/json"}
+
+    # capture response from API and parse out status code
+    response = requests.get(uri, headers=headers)
+    status_code = response.status_code
+
+    if status_code == 404:  # if workspace does not exist
+        return False
+    elif status_code == 200:  # if workspace already exists
+        print(f"Workspace already exists with name: {project}/{workspace_name}.")
+        print(f"Existing workspace details: {json.dumps(response.json(), indent=2)}")
+        # make user decide if they want to update/overwrite existing workspace
+        update_existing_ws = input("Would you like to continue modifying the existing workspace? (Y/N)" + "\n")
+        return update_existing_ws.upper()
+    else:  # if other error
+        return response.json()
+
+
+def create_workspace(json_request, workspace_name, project="anvil-datastorage"):
+    """Create the Terra workspace with given authorization domain."""
+
+    # check if workspace already exists
+    status_ws_exists = check_workspace_exists(workspace_name, project)
+
+    # if workspace does not exist, create workspace
+    if not status_ws_exists:
```

Comment:
this logic broke my brain a little... `status_ws_exists` could be False or a string or a json.... could you simplify `check_workspace_exists` to simply return True or False, and if it returns True, you prompt the user to continue or not inside THIS function? I think that would help with the logic flow